### DISCUSSION TITLE
Realtime + RLS: update a column that controls the policy

### DISCUSSION THREAD

**Discussion by Schniz** [supabase/supabase#7439] [Questions]
Hey there!

I've been thinking about it [for a while](https://twitter.com/galstar/status/1466318414453121027).
Is there a way to support updating data that affects the policy and get a `delete` event (or `updated`, at least)?

For instance, if I want to apply soft deletes, I might have a policy that filters out to `where deleted_at is not null`, and when I want to delete I will set `deleted_at = now()`. That would cause realtime to stop sending events for that specific row, therefore the client will never know the row got deleted.

It can work the same for permissions changes etc.

I wonder if there's such a way to model the data differently that will allow it to work.
I _guess_ a transaction of `delete` and an `insert` might work, but feels weird and can cause issues with indices 🤔 

Thanks in advance. Big fan of Supabase!

**Comment by w3b6x9** [comment]
@Schniz check out:

> Important Notes:
> 
> - Row level security is not applied to delete statements
> - The key/value pairs displayed in the old_record field include the table's identity columns for the record being updated/deleted. To display all values in old_record set the replica identity for the table to full
> - When a delete occurs, the contents of old_record will be broadcast to all subscribers to that table so ensure that each table's replica identity only contains information that is safe to expose publicly

- https://github.com/supabase/walrus

**Reply by Schniz** [reply]
my issues aren’t with deletes, but with updates. Updating a row that causes RLS to be unreachable for the current user feels like it should be the same as deleting

**Reply by Schniz** [reply]
I guess that a transaction with delete and insert would work for me
